test/e2e/examples.go

```diff
@@ -157,6 +159,109 @@ var _ = Describe("Examples e2e", func() {
 			}
 		})
 	})
+	Describe("[Example]ClusterDns", func() {
+		It("should create pod that uses dns", func() {
+			mkpath := func(file string) string {
+				return filepath.Join(testContext.RepoRoot, "examples/cluster-dns", file)
+			}
+
+			// contrary to the example, this test does not use contexts, for simplicity
+			// namespaces are passed directly.
+			// Also, for simplicity, we don't use yamls with namespaces, but we
+			// create testing namespaces instead.
+
+			backendRcYaml := mkpath("dns-backend-rc.yaml")
+			backendRcName := "dns-backend"
+			backendSvcYaml := mkpath("dns-backend-service.yaml")
+			backendSvcName := "dns-backend"
+			backendPodName := "dns-backend"
+			frontendPodYaml := mkpath("dns-frontend-pod.yaml")
+			frontendPodName := "dns-frontend"
+			frontendPodContainerName := "dns-frontend"
+
+			podOutput := "Hello World!"
+
+			// we need two namespaces anyway, so let's forget about
+			// the one created in BeforeEach and create two new ones.
+			namespaces := []*api.Namespace{nil, nil}
+			for i := range namespaces {
+				var err error
+				namespaces[i], err = createTestingNS(fmt.Sprintf("dnsexample%d", i), c)
+				if namespaces[i] != nil {
+					defer c.Namespaces().Delete(namespaces[i].Name)
+				}
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			for _, ns := range namespaces {
+				defer runKubectl("stop", "-f", backendRcYaml, getNsCmdFlag(ns))
```

Comment:
Actually, deleting namespaces is just enough. No need to delete everything from them before.